src/routes/dashboard/helpers/filter.ts

```diff
@@ -0,0 +1,98 @@
+import type { Contributor, Question, Questions } from '../types'
+
+function filterByDate(questions: Question[], dates: Date[]): Question[] {
+  return questions.filter(
+    (question) =>
+      new Date(question.createdAt) >= dates[0] &&
+      new Date(question.createdAt) < dates[1]
+  )
+}
+
+function filterByChannel(channels: string[], questions: Question[]) {
+  return questions.filter((question) => channels.includes(question.channelName))
+}
+
+/** counts the number of questions between two dates in each category */
+function filterQuestionsByDate(
+  datedQuestions: Map<string, Questions>,
+  questions: Questions,
+  dates: Date[]
+): Questions {
+  const filteredQs = {
+    total: [],
+    unanswered: [],
+    staff: [],
+    community: [],
+  } as Questions
+
+  Object.entries(questions).forEach(([category, categoryQuestions]) => {
+    const questionsBetweenDates =
+      categoryQuestions.filter(
```

Comment:
`Array.filter` _should_ return an empty array if nothing is found, however is this the same logic as `filterByDate`?

Reply:
oh yes good catch